modules/javafx.graphics/src/main/native-glass/mac/PlatformSupport.m

```diff
@@ -0,0 +1,281 @@
+    }
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.highlightColor" value:[NSColor highlightColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.shadowColor" value:[NSColor shadowColor]];
+
+    // Adaptable system colors
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemBlueColor" value:[NSColor systemBlueColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemBrownColor" value:[NSColor systemBrownColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemGrayColor" value:[NSColor systemGrayColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemGreenColor" value:[NSColor systemGreenColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemIndigoColor" value:[NSColor systemIndigoColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemOrangeColor" value:[NSColor systemOrangeColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemPinkColor" value:[NSColor systemPinkColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemPurpleColor" value:[NSColor systemPurpleColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemRedColor" value:[NSColor systemRedColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemTealColor" value:[NSColor systemTealColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemYellowColor" value:[NSColor systemYellowColor]];
+}
+
++ (void)updatePreferences:(jobject)application {
+    GET_MAIN_JENV;
+
+    jobject newPreferences = [self collectPreferences];
+    if (newPreferences == nil) {
+        return;
+    }
+
+    jboolean preferencesChanged = !(*env)->CallBooleanMethod(
+        env, newPreferences, jObjectEqualsMethod, currentPreferences);
+    GLASS_CHECK_EXCEPTION(env);
+
+    if (preferencesChanged) {
+        if (currentPreferences != nil) {
+            (*env)->DeleteGlobalRef(env, currentPreferences);
+        }
+
+        currentPreferences = (*env)->NewGlobalRef(env, newPreferences);
+
+        jobject unmodifiablePreferences = (*env)->CallStaticObjectMethod(
+            env, jCollectionsClass, jCollectionsUnmodifiableMapMethod, newPreferences);
+        GLASS_CHECK_EXCEPTION(env);
+
+        if (unmodifiablePreferences != nil) {
+            (*env)->CallVoidMethod(
+                env, application,
+                javaIDs.MacApplication.notifyPreferencesChanged,
+                unmodifiablePreferences);
+
+            (*env)->DeleteLocalRef(env, unmodifiablePreferences);
+        }
+
+    }
+
+    (*env)->DeleteLocalRef(env, newPreferences);
+}
+
++ (void)putString:(jobject)preferences key:(const char*)key value:(const char*)value {
+    GET_MAIN_JENV;
+
+    (*env)->CallObjectMethod(env, preferences, jMapPutMethod,
+        (*env)->NewStringUTF(env, key),
+        value != nil ? (*env)->NewStringUTF(env, value) : nil);
```

Comment:
You should check for pending exceptions (`GLASS_CHECK_EXCEPTION`) after this call, and maybe check for `NewStringUTF` returning null.